ntbs-service/Jobs/DataQualityAlertsJob.cs

```diff
@@ -34,63 +34,71 @@ public async Task Run(IJobCancellationToken token)
                 await _alertService.AddUniqueAlertAsync(alert);
             }
-
+            
             var notificationsForClusterAlerts =
                 await _dataQualityRepository.GetNotificationsEligibleForDataQualityClusterAlerts();
             foreach (var notification in notificationsForClusterAlerts)
             {
                 var alert = new DataQualityClusterAlert {NotificationId = notification.NotificationId};
                 await _alertService.AddUniqueAlertAsync(alert);
             }
-
+            
             var notificationsForTreatmentOutcome12Alerts =
                 await _dataQualityRepository.GetNotificationsEligibleForDataQualityTreatmentOutcome12Alerts();
             foreach (var notification in notificationsForTreatmentOutcome12Alerts)
             {
                 var alert = new DataQualityTreatmentOutcome12 {NotificationId = notification.NotificationId};
                 await _alertService.AddUniqueAlertAsync(alert);
             }
-
+            
             var notificationsForTreatmentOutcome24Alerts =
                 await _dataQualityRepository.GetNotificationsEligibleForDataQualityTreatmentOutcome24Alerts();
             foreach (var notification in notificationsForTreatmentOutcome24Alerts)
             {
                 var alert = new DataQualityTreatmentOutcome24 {NotificationId = notification.NotificationId};
                 await _alertService.AddUniqueAlertAsync(alert);
             }
-
+            
             var notificationsForTreatmentOutcome36Alerts =
                 await _dataQualityRepository.GetNotificationsEligibleForDataQualityTreatmentOutcome36Alerts();
             foreach (var notification in notificationsForTreatmentOutcome36Alerts)
             {
                 var alert = new DataQualityTreatmentOutcome36 {NotificationId = notification.NotificationId};
                 await _alertService.AddUniqueAlertAsync(alert);
             }
-
+            
             var notificationsForDotVotAlerts =
                 await _dataQualityRepository.GetNotificationsEligibleForDotVotAlerts();
-            foreach (var notification in notificationsForDotVotAlerts) 
+            foreach (var notification in notificationsForDotVotAlerts)
             {
                 var alert = new DataQualityDotVotAlert {NotificationId = notification.NotificationId};
                 await _alertService.AddUniqueAlertAsync(alert);
             }
 
+            var possibleDuplicateNotifications =
+                await _dataQualityRepository.GetNotificationIdsEligibleForPotentialDuplicateAlerts();
+            foreach (var notification in possibleDuplicateNotifications)
+            {
+                var alert = new DataQualityPotentialDuplicateAlert {NotificationId = notification.NotificationId, DuplicateId = notification.DuplicateId};
+                await _alertService.AddUniquePotentialDuplicateAlertAsync(alert);
+            }
+
```

Comment:
Given what we said about the memory footprint being one of the problems for these jobs, do we want to break it out into a separate job (it can be queued at the same time, it'll just get picked up in "some" order)?